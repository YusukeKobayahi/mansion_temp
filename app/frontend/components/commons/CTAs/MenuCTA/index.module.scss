@import "~/styles/variables";
@import "~/styles/utils";

.container {
  width: 400px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-left: auto;
}

.btn {
  flex: auto;
  border: 2px solid $color-main;
  color: $color-main;
  background-color: #ffffff;
  text-align: center;
  padding: 10px 12px;
  cursor: pointer;
  box-shadow: 0 4px 4px $color-main;
  font-size: 16px;

  &:nth-child(n + 2) {
    margin-left: 10px;
  }
}
