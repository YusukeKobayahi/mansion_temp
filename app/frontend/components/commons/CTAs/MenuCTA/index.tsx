import styles from "./index.module.scss";
import { clickMembers } from "~/lib/utils";
import classnames from "classnames";
import { GoogleAnalyticsEvent } from "~/lib/types";

const MenuCTA: React.FC = () => {
  const clickCTA = (label: GoogleAnalyticsEvent["label"]) => {
    clickMembers({
      action: "msl_cta_click",
      category: "msl_ichiran",
      label: label,
      value: 1,
    });
  };
  return (
    <section className={classnames(styles.container, "pcOnly")}>
      <button className={styles.btn} onClick={() => clickCTA("msl_lp_mikokai")}>
        未公開物件の
        <br />
        情報が欲しい
      </button>
      <button className={styles.btn} onClick={() => clickCTA("msl_lp_gensen")}>
        複数の不動産会社
        <br />
        に相談したい
      </button>
    </section>
  );
};

export default MenuCTA;
