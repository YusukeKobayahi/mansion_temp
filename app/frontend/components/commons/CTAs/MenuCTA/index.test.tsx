import { render } from "@testing-library/react";
import MenuCTA from ".";

describe("components/commons/CTAs/CTAs", () => {
  it("正常に表示される", () => {
    const { asFragment } = render(<MenuCTA />);
    expect(asFragment()).toMatchSnapshot();
  });
});
