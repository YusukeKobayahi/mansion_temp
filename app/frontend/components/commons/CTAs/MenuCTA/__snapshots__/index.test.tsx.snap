// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/CTAs/CTAs 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container pcOnly"
  >
    <button
      class="btn"
    >
      未公開物件の
      <br />
      情報が欲しい
    </button>
    <button
      class="btn"
    >
      複数の不動産会社
      <br />
      に相談したい
    </button>
  </section>
</DocumentFragment>
`;
